src/ray/core_worker/core_worker.cc

```diff
@@ -489,11 +489,24 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
       new CoreWorkerDirectActorTaskSubmitter(core_worker_client_pool_, memory_store_,
                                              task_manager_));
 
+  auto node_addr_factory = [this](const NodeID &node_id) {
+    absl::optional<rpc::Address> addr;
+    if (auto node_info = gcs_client_->Nodes().Get(node_id)) {
+      rpc::Address address;
+      address.set_raylet_id(node_info->node_id());
+      address.set_ip_address(node_info->node_manager_address());
+      address.set_port(node_info->node_manager_port());
+      addr = address;
+    }
+    return addr;
+  };
+  std::shared_ptr<LessorPickerInterface> lessor_picker =
+      std::make_shared<LessorPicker>(reference_counter_, node_addr_factory);
   direct_task_submitter_ =
       std::unique_ptr<CoreWorkerDirectTaskSubmitter>(new CoreWorkerDirectTaskSubmitter(
           rpc_address_, local_raylet_client_, core_worker_client_pool_,
-          raylet_client_factory, memory_store_, task_manager_, local_raylet_id,
-          RayConfig::instance().worker_lease_timeout_milliseconds(),
+          raylet_client_factory, std::move(lessor_picker), memory_store_, task_manager_,
```

Comment:
For more convenient unit testing, could you pass in nullptr here if the feature flag is off, instead of checking it within the `CoreWorkerDirectTaskSubmitter` class?

Reply:
Sure! Although would it make more sense to provide a lessor picker that always picks the local node for the first lease (when the feature flag is off) instead of a `nullptr`? With the way things are currently structured, the lessor picker could always return `absl::nullopt` and the `CoreWorkerDirectTaskSubmitter` would fall back to the local node, or I could move that fallback logic into the lessor picker (I've already been tempted to do that). I'm thinking that would be nicer than having to check for `nullptr` or the feature flag in the `CoreWorkerDirectTaskSubmitter`.